### PR TITLE
Update to the current Cypress Eslint plugin

### PR DESCRIPTION
**Pull request by bahmutov**
- lint blog recipes using https://github.com/cypress-io/eslint-plugin-dev

## Note

Some projects crash the linter

### CODE REVIEW

**Comment by bahmutov** [comment]
Example recipe that crashes the linter with error `Cannot read property 'push' of undefined`

```
$ DEBUG=* npx eslint examples/blogs__application-actions
...

  eslint:cli-engine Lint /Users/gleb/git/cypress-example-recipes/examples/blogs__application-actions/vendor/js/react-with-addons.js +250ms
  eslint:linter Linting code for /Users/gleb/git/cypress-example-recipes/examples/blogs__application-actions/vendor/js/react-with-addons.js (pass 1) +5ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/gleb/git/cypress-example-recipes/examples/blogs__application-actions/vendor/js/react-with-addons.js +0ms
  json-settings user supplied setting: sort-package-json +250ms
  json { extension: '.js' } +250ms
  json { mode: undefined } +0ms
  eslint:config-comment-parser Parsing String config +564ms
  eslint:config-comment-parser Parsing list config +0ms
  eslint:config-comment-parser Parsing list config +0ms
  eslint:config-comment-parser Parsing String config +0ms
Cannot read property 'push' of undefined
```

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+examples/blogs__application-actions
+examples/blogs__e2e-snapshots/snapshots.js
+examples/blogs__element-coverage/src
+examples/preprocessors__typescript-webpack/dist
+node_modules
```
Comment (Bkucera) [inline .eslintignore]: this should fix at least the error you mentioned
```suggestion
node_modules
**/vendor
```

**Comment by Bkucera** [comment]
@bahmutov do you want to enable json linting as well?

**Comment by bahmutov** [comment]
yes, how do I enable JSON linting?

**Comment by bahmutov** [comment]
Merging this in to avoid overwhelming PR with so many whitespace changes and getting https://github.com/cypress-io/cypress-example-recipes/pull/387 fixed

**Comment by Bkucera** [comment]
ah, missed it by just a little

**Pull request merged**